### PR TITLE
add signal traps for INT and TERM for followers

### PR DESCRIPTION
**Pull request by viktorerlingsson**
### WHAT is this pull request doing?
Adds signal traps for INT and TERM to properly shut down follower and close sockets. 

Without this I end up with extra unix & http sockets when closing a follower with ctrl+c. 

### HOW can this pull request be tested?
Manual

### CODE REVIEW

File: src/lavinmq/clustering/client.cr [hunk 1]
```diff
@@ -33,6 +33,18 @@ module LavinMQ
           @unix_amqp_proxy = Proxy.new(@config.unix_path) unless @config.unix_path.empty?
           @unix_http_proxy = Proxy.new(@config.http_unix_path) unless @config.http_unix_path.empty?
         end
+        setup_signal_traps
+      end
+
+      private def setup_signal_traps
+        Signal::INT.trap { close_and_exit }
+        Signal::TERM.trap { close_and_exit }
+      end
```
Comment (carlhoerberg) [inline src/lavinmq/clustering/client.cr:42]:
```suggestion
        Signal::INT.trap { close_and_exit }
        Signal::TERM.trap { close_and_exit }
      end
```

File: src/lavinmq/clustering/client.cr [hunk 2]
```diff
@@ -33,6 +33,18 @@ module LavinMQ
...
+
+      def close_and_exit
```
Comment (carlhoerberg) [inline src/lavinmq/clustering/client.cr:44]:
```suggestion
      private def close_and_exit
```

**Review by carlhoerberg — COMMENTED** [review]
How come sockets was "left over"? If the process exits all sockets opened by that process is closed as well?

**Comment by viktorerlingsson** [comment]
> How come sockets was "left over"? If the process exits all sockets opened by that process is closed as well?

Yeah, I still had the "files" for `lavinmq.sock` and `lavinmq-http.sock` after closing a follower with ctrl+c, and could not restart the follower unless I removed them. Would they remain if the sockets weren't properly unbound or something?

File: src/lavinmq/clustering/client.cr (hunk 1 above)
Comment (viktorerlingsson) [reply src/lavinmq/clustering/client.cr:42]: Sure, I just copied from launcher :)

**Comment by carlhoerberg** [comment]
I see, you need to try to delete the file before just like in `Launcher`: https://github.com/cloudamqp/lavinmq/blob/6bb3de850c37854272af5a49d31c5382d07b5e1d/src/lavinmq/server.cr#L182

Can happen if the clustering client was killed (eg. OOM), even with this patch.

**Review by kickster97 — APPROVED** [review]

**Pull request merged**